packages/edit-site/src/components/save-button/index.js

```diff
@@ -78,7 +81,27 @@ export default function SaveButton( {
 		}
 		return __( 'Save' );
 	};
-	const label = getLabel();
+
+	/**
+	 * We focus on adding the customization to the SaveButton's `onClick` and `label` for now.
+	 * We will provide the customization to the other entry points (e.g., SavePanel, SaveHub) in the future if needed.
+	 * @see https://github.com/WordPress/gutenberg/pull/56807
+	 */
+	const onClick = useCallback( () => {
+		const callback = applyFilters( 'edit-site.SaveButton.onClick', () =>
+			setIsSaveViewOpened( true )
+		);
+		callback();
```

Comment:
I think we can just return the function but it's optional 🙂

Reply:
I like this idea too.